packages/gatsby/cache-dir/loader.js

```diff
@@ -290,8 +291,18 @@ export class BaseLoader {
           }
           // undefined if final result is an error
           return pageResources
-        }
-      )
+        })
+
+      let componentChunkPromise
+      if (
+        process.env.NODE_ENV === `development` &&
+        process.env.GATSBY_EXPERIMENTAL_LAZY_DEVJS &&
+        options.isPrefetch === true
+      ) {
```

Comment:
I would make the remainder of this function its own function and conditionally call it. (In other words, it might make more sense to do an early return here).